torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py

```diff
@@ -65,3 +66,33 @@ def decompress(fut):
         return [decompressed_tensor]
 
     return fut.then(decompress)
+
+def fp16_compress_wrapper(hook: Callable[..., torch.futures.Future]) -> Callable[..., torch.futures.Future]:
+    """
+    This wrapper casts the input gradient tensors of a given DDP communication hook to half-precision
+    floating point format (torch.float16), and casts the resulting tensors of the given hook back to
```

Comment:
Nit: Please quote the data types with two grave accents (``).

``torch.float16``
``float32``